tsuru/client/deploy.go

```diff
@@ -248,17 +251,23 @@ func (c *AppDeploy) Run(context *cmd.Context, client *cmd.Client) error {
 	defer resp.Body.Close()
 	c.eventID = resp.Header.Get("X-Tsuru-Eventid")
 	c.m.Unlock()
-	scanner := bufio.NewScanner(resp.Body)
-	for scanner.Scan() {
+	var readBuffer [deployOutputBufferSize]byte
+	var readErr error
+	for readErr == nil {
+		var read int
+		read, readErr = resp.Body.Read(readBuffer[:])
 		c.m.Lock()
-		_, err := respBody.Write([]byte(scanner.Text() + "\n"))
-		if err != nil {
-			fmt.Fprintln(context.Stderr, "writing response:", err)
-		}
+		written, writeErr := respBody.Write(readBuffer[:read])
```

Comment:
Shouldn't you check if `read > 0` before writing?

Reply:
Sure, better to avoid locking for no reason, even though calling the writer with an empty buffer won't cause an error.